src-tauri/src/main.rs

```diff
@@ -85,66 +82,70 @@ async fn main() {
+    info!("Running the macro backend");
+    if let Err(e) = backend.init().await {
+        eprintln!("Initialization error: {}", e);
+    };
 
+    // Read the options from the config.
     let set_autolaunch: bool = backend.config.read().await.auto_start;
     let set_launch_minimized: bool = backend.config.read().await.minimize_at_launch;
 
-    // Begin the main event loop. This loop cannot run on another thread on MacOS.
+    // Set the window behaviors (add buttons to the right click tray menu).
     let quit = CustomMenuItem::new("quit".to_string(), "Quit");
     let hide_show = CustomMenuItem::new("hide_show".to_string(), "Hide");
-
     let tray_menu = SystemTrayMenu::new()
         .add_item(hide_show)
         .add_native_item(SystemTrayMenuItem::Separator)
         .add_item(quit);
-
     let system_tray = SystemTray::new().with_menu(tray_menu);
 
+    // Initialize the main application
     tauri::Builder::default()
-        // This is where you pass in your commands
+        // State management is initialized
         .manage(backend)
+        // This is where commands shared with frontend are passed
         .invoke_handler(tauri::generate_handler![
             get_macros,
             set_macros,
             get_config,
             set_config,
-            control_grabbing,
-            get_monitor_data
+            control_grabbing
         ])
         .setup(move |app| {
             let app_name = &app.package_info().name;
-            let current_exe = current_exe().unwrap();
+            if let Ok(current_exe) = current_exe() {
+                let auto_start = auto_launch::AutoLaunchBuilder::new()
+                    .set_app_name(app_name)
+                    .set_app_path(current_exe.as_path().to_str().unwrap())
+                    .set_use_launch_agent(true)
+                    .build()
+                    .expect("App name is empty, or unsupported OS is used.");
 
-            let auto_start = auto_launch::AutoLaunchBuilder::new()
-                .set_app_name(app_name)
-                .set_app_path(current_exe.as_path().to_str().unwrap())
-                .set_use_launch_agent(true)
-                .build()
-                .unwrap();
-
-            match set_autolaunch {
-                true => auto_start.enable().unwrap(),
-                false => {
-                    if let Err(e) = auto_start.disable() {
-                        match e {
-                            auto_launch::Error::Io(err) => match err.kind() {
-                                std::io::ErrorKind::NotFound => {
-                                    info!("Autostart is already removed, finished checking.")
-                                }
-                                _ => error!("{}", err),
-                            },
-                            _ => error!("{}", e),
+                match set_autolaunch {
+                    true => auto_start.enable().expect("Registry key failed to write."),
```

Comment:
If you're gracefully handling errors here, then you should be doing it for the `auto_start.enable()`. It doesn't really make sense to panic if it can't enable. It should log if it was successful, if it wasn't, then you should do an error log

Reply:
Tried to make the behavior consistent: [5243083](https://github.com/WootingKb/wooting-macros/pull/194/commits/52430832490a73c9d784ec8058f8ed5b21337590)